source/obstacle/ObstacleGenerator.hx

```diff
@@ -7,73 +7,71 @@ import flixel.util.FlxTimer;
 import obstacle.Obstacle;
 
 @:generic
-class ObstacleGenerator<T:Obstacle> extends FlxBasic
-{
+class ObstacleGenerator<T:Obstacle> extends FlxBasic {
 	private var spawnTimer:FlxTimer = new FlxTimer();
 
 	public var obstacles(default, null):FlxTypedGroup<T>;
 	public var baseObstacleParameters:ObstacleParameters;
 	public var variation:ObstacleVariation;
 	public var spawnDelay(default, set):Float;
 
-	public function new(spawnDelay:Float, baseObstacleParameters:ObstacleParameters,
-			variation:ObstacleVariation, obstacles:FlxTypedGroup<T>)
-	{
+	public function new(spawnDelay:Float, baseObstacleParameters:ObstacleParameters, variation:ObstacleVariation, obstacles:FlxTypedGroup<T>) {
+		// call parent constructor
 		super();
 
+		// set to the obstacle objects
 		this.obstacles = obstacles;
+		// the base parameters used to create the obstacles
 		this.baseObstacleParameters = baseObstacleParameters;
+		// the variation that will be applied to the obstacles
 		this.variation = variation;
+		// the delay between obstacle spawning
 		this.spawnDelay = spawnDelay;
-
+		// start the spawn timer
+		// used to spawn in an obstacle
 		spawnTimer.start(spawnDelay, generateObstacle, 0);
 	}
 
-	public function generateObstacle(timer:FlxTimer)
-	{
+	public function generateObstacle(timer:FlxTimer) {
 		var recycled:Obstacle = obstacles.recycle();
 		recycled.init(variation.apply(baseObstacleParameters));
 	}
 
-	public function set_spawnDelay(spawnDelay:Float):Float
-	{
+	public function set_spawnDelay(spawnDelay:Float):Float {
 		this.spawnDelay = spawnDelay;
-		if (spawnTimer.active == true)
-		{
+		if (spawnTimer.active == true) {
 			spawnTimer.time = this.spawnDelay;
 		}
 		return this.spawnDelay;
 	}
 }
 
-class ObstacleVariation
-{
+class ObstacleVariation {
 	public var y:Float;
 	public var speed:Float;
 	public var width:Float;
 	public var height:Float;
 
-	public function new(y:Float = 0, speed:Float = 0, width:Float = 0, height:Float = 0)
-	{
+	public function new(y:Float = 0, speed:Float = 0, width:Float = 0, height:Float = 0) {
 		this.y = y;
 		this.speed = speed;
 		this.width = width;
 		this.height = height;
 	}
 
-	public inline function apply(base:ObstacleParameters):ObstacleParameters
-	{
+	public inline function apply(base:ObstacleParameters):ObstacleParameters {
+		// creates a new obstacle parameter obj that has different parametes than it
+		// used to
 		var x = base.x;
-		var y = randomize(base.y, this.y);
-		var speed = randomize(base.speed, this.speed);
+		var speed = base.speed;
 		var width = Std.int(randomize(base.width, this.width));
 		var height = Std.int(randomize(base.height, this.height));
-
+		var y = Std.int(FlxG.random.float(30, FlxG.height));
```

Comment:
Since this code replaces how the returned `y` is generated to not involve the `ObstacleVariation` object's `y` at all, let's just remove the `y` field from `ObstacleVariation` completely and take it out of the constructor.